### PR TITLE
Feature/si 2500 add nonce to the meta transactions

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

File: Selectors.md [hunk 1]
```diff
@@ -433,6 +433,12 @@
 |-|-|
 | 0xc9134785 | UintUtils__InsufficientHexLength() |
 
+## Nonces
+#### Functions
+| Selector | Signature |
+|-|-|
+| 0x86e2b63f | nonces(bytes32,address) |
```
Comment (elffjs) [inline Selectors.md:440]: Maybe this should be singular?

File: contracts/implementations/MultipleMinter.sol [hunk 2]
```diff
@@ -46,8 +49,6 @@ contract MultipleMinter is
     function mintVehicleAndSdSign(
         MintVehicleAndSdInput calldata data
     ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
-        NodesStorage.Storage storage ns = NodesStorage.getStorage();
```
Comment (elffjs) [inline contracts/implementations/MultipleMinter.sol:49]: Cutting down on variables. I see.

File: contracts/implementations/nodes/AftermarketDevice.sol [hunk 3]
```diff
@@ -258,24 +259,45 @@ contract AftermarketDevice is
...
+            )
+        );
+        bytes32 messageDevice = keccak256(
+            abi.encode(
+                CLAIM_TYPEHASH,
+                aftermarketDeviceNode,
+                owner,
+                NoncesInternal._useNonce(
+                    CLAIM_TYPEHASH,
+                    aftermarketDeviceAddress
```
Comment (elffjs) [inline contracts/implementations/nodes/AftermarketDevice.sol:280]: My bad on this. I think we need to do one nonce, the user’s, for now. As long as the message contains both the user’s address and the user’s nonce, I think we are okay.

File: contracts/implementations/MultipleMinter.sol [hunk 4]
```diff
@@ -27,7 +28,9 @@ contract MultipleMinter is
     SyntheticDeviceInternal
 {
     bytes32 private constant MINT_VEHICLE_SD_TYPEHASH =
-        keccak256("MintVehicleAndSdSign(uint256 integrationNode)");
+        keccak256(
```
Comment (elffjs) [inline contracts/implementations/MultipleMinter.sol:31]: For synthetics we have a lot of control over the device wallet, so we could do this.

But we could get away with—is there a reason this doesn’t include the owner’s address? Then I’d feel good about using the user nonce, as for aftermarket.

Comment (elffjs) [reply contracts/implementations/MultipleMinter.sol:31]: I guess for synthetics they can only ever get minted once. Although I guess you could re-use a signature from a burned one, so maybe that’s not true.

File: contracts/implementations/MultipleMinter.sol (hunk 2 above)
Comment (LorranSutter) [reply contracts/implementations/MultipleMinter.sol:49]: I wonder how long I'll be able to do this

File: Selectors.md (hunk 1 above)
Comment (elffjs) [reply Selectors.md:440]: Alright.

**Review by elffjs — APPROVED** [review]

**Pull request merged**